fw/http_parser.c

```diff
@@ -2265,45 +2359,59 @@ __req_parse_cache_control(TfwHttpReq *req, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(Req_I_CC_n) {
+		if (unlikely(!parser->cache_control.dir_allowed))
+			__FSM_EXIT(TFW_BLOCK);
 		TRY_STR_LAMBDA("no-cache", {
-			parser->_acc = TFW_HTTP_CC_NO_CACHE;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_CACHE;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_LAMBDA("no-store", {
-			parser->_acc = TFW_HTTP_CC_NO_STORE;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_STORE;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_LAMBDA("no-transform", {
-			parser->_acc = TFW_HTTP_CC_NO_TRANSFORM;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_TRANSFORM;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_INIT();
 		__FSM_I_JMP(Req_I_CC_Ext);
 	}
 
 	__FSM_STATE(Req_I_CC_o) {
+		if (unlikely(!parser->cache_control.dir_allowed))
+			__FSM_EXIT(TFW_BLOCK);
 		TRY_STR_LAMBDA("only-if-cached", {
-			parser->_acc = TFW_HTTP_CC_OIFCACHED;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_OIFCACHED;
 		}, Req_I_CC_o, Req_I_CC_Flag);
 		TRY_STR_INIT();
 		__FSM_I_JMP(Req_I_CC_Ext);
 	}
 
 	__FSM_STATE(Req_I_CC_Flag) {
-		WARN_ON_ONCE(!parser->_acc);
+		/* A start of a common flag successfully detected */
```

Comment:
What is a "common flag"? The comment is very confusing in context of the state name `Req_I_CC_Flag`, where "flag" is about the parser flag, see https://github.com/tempesta-tech/tempesta/pull/1539/files#r774807540